xoinvader/animation.py

```diff
@@ -126,17 +141,71 @@ def name(self):
         """
         return self._name
 
-    def update(self):
-        """Update animation state.
+    def _apply_value(self, value):
+        """Apply new value to linked object.
+
+        :param obj value: value to apply
+        """
+
+        setattr(self._obj, self._attr, value)
+
+    def _update_interpolated(self):
+        """Advance animation and interpolate value.
+
+        NOTE: animation frame switching depends on interp mode
+        animation with interpolation switches frame only when current local
+        time exceeds NEXT frames' time border.
+        """
+
+        self._check_animation_state()
+        self._timer.update()
+
+        current_time = self._timer.get_elapsed()
+        keyframe = self._keyframes[self._current]
+        next_keyframe = self._keyframes[self._current + 1]
```

Comment:
Will it raise proper exception when there's no next keyframe?

And will it be handled properly?